http-nu: init at 0.5.0

cboecking opened the pull request:
## Description

Adds http-nu, a tool to serve Nushell closures over HTTP.

## Testing

- [x] Built on x86_64-linux
- [x] Tested `http-nu --version` (returns: http-nu 0.5.0)
- [x] Tested `http-nu --help` (displays usage information)
- [x] Binary executes successfully

## Notes

- Tests are disabled (`doCheck = false`) as they fail in the Nix sandbox due to file system and network access requirements
- This is a new package with no reverse dependencies
- Upstream project: https://github.com/cablehead/http-nu
- License: MIT

cboecking committed:
http-nu: init at 0.5.0

iedame commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 458947`
Commit: [`1642eacf725420005e2cf9c9ed39dfdbdcabca70`](https://github.com/NixOS/nixpkgs/commit/1642eacf725420005e2cf9c9ed39dfdbdcabca70) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/1642eacf725420005e2cf9c9ed39dfdbdcabca70..pull/458947/head))
Merge: [`15c6bd2c2dfbe61ea65578a0ba9716e503ecf53d`](https://github.com/NixOS/nixpkgs/commit/15c6bd2c2dfbe61ea65578a0ba9716e503ecf53d)

Logs: https://github.com/iedame/nixpkgs-review-gha/actions/runs/19130351941


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>http-nu</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>http-nu</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>http-nu</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>http-nu</summary>
<pre>   Compiling sys-locale v0.3.2
   Compiling erased-serde v0.4.8
   Compiling zeroize v1.8.2
   Compiling fs_extra v1.3.0
   Compiling dunce v1.0.5
   Compiling nu-utils v0.108.0
   Compiling aws-lc-sys v0.32.3
   Compiling dirs-sys v0.5.0
   Compiling objc2-io-kit v0.3.2
   Compiling itertools v0.14.0
error: failed to run custom build command for `libproc v0.14.11`

Caused by:
  process didn&#x27;t exit successfully: `/nix/build/nix-7054-1758958356/source/target/release/build/libproc-a9166c386465ef44/build-script-build` (exit status: 101)
  --- stderr

  thread &#x27;main&#x27; panicked at /nix/build/nix-7054-1758958356/http-nu-0.5.0-vendor/bindgen-0.72.1/lib.rs:616:27:
  Unable to find libclang: &quot;couldn&#x27;t find any valid shared libraries matching: [&#x27;libclang.dylib&#x27;], set the `LIBCLANG_PATH` environment variable to a path where one of these files can be found (invalid: [])&quot;
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>http-nu</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>http-nu</summary>
<pre>   Compiling ref-cast v1.0.25
   Compiling zeroize v1.8.2
   Compiling erased-serde v0.4.8
   Compiling owo-colors v4.2.3
   Compiling dunce v1.0.5
   Compiling sys-locale v0.3.2
   Compiling nu-utils v0.108.0
   Compiling aws-lc-sys v0.32.3
   Compiling dirs-sys v0.5.0
   Compiling objc2-io-kit v0.3.2
error: failed to run custom build command for `libproc v0.14.11`

Caused by:
  process didn&#x27;t exit successfully: `/nix/build/nix-6257-4073133899/source/target/release/build/libproc-e49be16676325bae/build-script-build` (exit status: 101)
  --- stderr

  thread &#x27;main&#x27; panicked at /nix/build/nix-6257-4073133899/http-nu-0.5.0-vendor/bindgen-0.72.1/lib.rs:616:27:
  Unable to find libclang: &quot;couldn&#x27;t find any valid shared libraries matching: [&#x27;libclang.dylib&#x27;], set the `LIBCLANG_PATH` environment variable to a path where one of these files can be found (invalid: [])&quot;
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...</pre>
</details>
</details>

iedame commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 458947`
Commit: [`23bd834843af21ce8062d6f768b7f80d9d39760a`](https://github.com/NixOS/nixpkgs/commit/23bd834843af21ce8062d6f768b7f80d9d39760a) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/23bd834843af21ce8062d6f768b7f80d9d39760a..pull/458947/head))
Merge: [`5a5601c09e024fe50471d5b80af5756c3fcc591b`](https://github.com/NixOS/nixpkgs/commit/5a5601c09e024fe50471d5b80af5756c3fcc591b)

Logs: https://github.com/iedame/nixpkgs-review-gha/actions/runs/19178223179


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>http-nu</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>http-nu</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>http-nu</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>http-nu</li>
  </ul>
</details>

yzhou216 commented on pkgs/by-name/ht/http-nu/package.nix:
```diff
@@ -0,0 +1,50 @@
+  stdenv,
+}:
+
+rustPlatform.buildRustPackage rec {
```
```suggestion
rustPlatform.buildRustPackage (finalAttrs: {
```

Use `finalAttrs` pattern.

yzhou216 commented on pkgs/by-name/ht/http-nu/package.nix:
```diff
@@ -0,0 +1,50 @@
+    mainProgram = "http-nu";
+    platforms = lib.platforms.unix;
+  };
+}
```
```suggestion
})
```

yzhou216 commented on pkgs/by-name/ht/http-nu/package.nix:
```diff
@@ -0,0 +1,50 @@
+      capabilities.
+    '';
+    homepage = "https://github.com/cablehead/http-nu";
+    changelog = "https://github.com/cablehead/http-nu/blob/v${version}/changes/v${version}.md";
```
```suggestion
    changelog = "https://github.com/cablehead/http-nu/blob/v${finalAttrs.version}/changes/v${finalAttrs.version}.md";
```

yzhou216 reviewed (changes requested):
Welcome!

yzhou216 reviewed (commented):
Adding maintainers should be split up to two different commits, one maintainer per commit.

cboecking committed:
maintainers: add cablehead

cboecking committed:
maintainers: add cboecking

cboecking commented:
Thanks for the feedback! I've split the maintainer additions into separate commits as requested:

- `568ddd43` - maintainers: add cablehead
- `b3442d64` - maintainers: add cboecking

Ready for re-review.

cboecking commented:
Thanks for the thorough review @yzhou216! I've addressed all your feedback:

**Maintainer commits:**
- ✅ Split into separate commits (one per maintainer)
  - `a378b1a` - maintainers: add cablehead
  - `e0894c85` - maintainers: add cboecking

**Code improvements (squashed into init commit `fedf50e9`):**
- ✅ Use `finalAttrs` pattern for `buildRustPackage`
- ✅ Change `stdenv` to `stdenvNoCC`
- ✅ Use `libclang` from top-level instead of `llvmPackages`
- ✅ Use `tag` instead of `rev` in `fetchFromGitHub`
- ✅ Fix libclang setup with proper `nativeBuildInputs` and `env` structure
- ✅ Use `finalAttrs.version` in changelog URL
- ✅ Remove `platforms` line (handled by `buildRustPackage`)
- ✅ Improved test disabling comment with investigation details

**Testing:**
Regarding selective test disabling: I investigated the test suite and found that most tests require sandbox-incompatible operations (network access for integration tests, filesystem writes for handler tests). Only `test_engine` could run in sandbox, but selective enablement would add maintenance burden with limited value. I've updated the comment to document this.

Package builds successfully on x86_64-linux. Ready for re-review!

cboecking commented:
@yzhou216 - I believe I have done everything.

yzhou216 reviewed (commented):
Please rebase and order the commits to be:

```
maintainers: add cablehead
maintainers: add cboecking
http-nu: init at 0.5.0
```

Squash `http-nu: add maintainers` into `http-nu: init at 0.5.0`

cboecking commented:
Rebased and reordered commits as requested. The commits are now:
1. maintainers: add cablehead
2. maintainers: add cboecking
3. http-nu: init at 0.5.0 (with maintainers squashed in)

yzhou216 commented on pkgs/by-name/ht/http-nu/package.nix:
```diff
@@ -0,0 +1,52 @@
+  # - test_handler requires temporary filesystem writes
+  # - Only test_engine could run in sandbox, but selective enablement adds maintenance burden
+  doCheck = false;
+
```
```suggestion

  doInstallCheck = true;
  nativeInstallCheckInputs = [ versionCheckHook ];
  versionCheckProgramArg = "--version";

  passthru.updateScript = nix-update-script { };

```

yzhou216 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 458947`
Commit: [`1829e71e8637b5d756884393539a69a248ada66c`](https://github.com/NixOS/nixpkgs/commit/1829e71e8637b5d756884393539a69a248ada66c) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/1829e71e8637b5d756884393539a69a248ada66c..pull/458947/head))
Merge: [`4dcd682cd08aa439f0514c59f0260cebb4363d6b`](https://github.com/NixOS/nixpkgs/commit/4dcd682cd08aa439f0514c59f0260cebb4363d6b)

Logs: https://github.com/yzhou216/nixpkgs-review-gha/actions/runs/19557734574

<details><summary>Download packages from cache:</summary><ul>
<li><details><summary><code>x86_64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-458947-x86_64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://yzhou216.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  yzhou216.cachix.org-1:O7nGwQnLGGBGE8zYFMipVzPVN956FcV57y9SqwEP+O8=
  ' \
  /nix/store/9lwvwl2paay8h69n9vfgkln1mbnp65i1-http-nu-0.5.0
```
</details></li>
<li><details><summary><code>aarch64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-458947-aarch64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://yzhou216.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  yzhou216.cachix.org-1:O7nGwQnLGGBGE8zYFMipVzPVN956FcV57y9SqwEP+O8=
  ' \
  /nix/store/w2nn6x1krhi4armawvc9hppl8j63fp1p-http-nu-0.5.0
```
</details></li>
<li><details><summary><code>x86_64-darwin</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-458947-x86_64-darwin \
  --option binary-caches 'https://cache.nixos.org/ https://yzhou216.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  yzhou216.cachix.org-1:O7nGwQnLGGBGE8zYFMipVzPVN956FcV57y9SqwEP+O8=
  ' \
  /nix/store/cz3b7qd0ja14rs96r0vbhxb0zwbb7wkh-http-nu-0.5.0
```
</details></li>
<li><details><summary><code>aarch64-darwin</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-458947-aarch64-darwin \
  --option binary-caches 'https://cache.nixos.org/ https://yzhou216.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  yzhou216.cachix.org-1:O7nGwQnLGGBGE8zYFMipVzPVN956FcV57y9SqwEP+O8=
  ' \
  /nix/store/75j7q0vnbnkla9kgkns5sadl3p1fvqdq-http-nu-0.5.0
```
</details></li>
</ul></details>


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>http-nu</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>http-nu</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>http-nu</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>http-nu</li>
  </ul>
</details>

cboecking commented:
Thanks @Aleksanaa! Applied your suggestion - now using `rustPlatform.bindgenHook` instead of manual libclang setup.

Aleksanaa commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 458947`
Commit: [`182ca73786c0ae0e5e747d55c2fc100d78a74ba8`](https://github.com/NixOS/nixpkgs/commit/182ca73786c0ae0e5e747d55c2fc100d78a74ba8) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/182ca73786c0ae0e5e747d55c2fc100d78a74ba8..pull/458947/head))
Merge: [`721f805fb7ce981e4cc5b0aa39329237f714abfc`](https://github.com/NixOS/nixpkgs/commit/721f805fb7ce981e4cc5b0aa39329237f714abfc)

Logs: https://github.com/Aleksanaa/nixpkgs-review-gha/actions/runs/19865736653


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>http-nu</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>http-nu</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>http-nu</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>http-nu</li>
  </ul>
</details>
